### ISSUE TITLE
let_underscore_untyped with type `_`

### ISSUE THREAD

**Issue by dtolnay** [rust-lang/rust-clippy#10441]
### Summary

In light of #10411, it seems likely that folks would work around this lint by replacing `let _ = ...` with `let _: _ = ...`.

However, that has _all_ of the same downsides described in #6842. It's just a noisier / less clear way of writing the same thing. It would be better to require `#[allow(clippy::let_underscore_untyped)]` when someone wants to disregard this lint, rather than `let _: _`.

### Lint Name

let_underscore_untyped

### Reproducer

```rust
#![deny(clippy::pedantic)]

fn f() -> impl Fn() {
    || {}
}

fn main() {
    let _: _ = f();
}
```

Should trigger let_underscore_untyped, but does not.

### Version

```text
rustc 1.69.0-nightly (13471d3b2 2023-03-02)
binary: rustc
commit-hash: 13471d3b2046cce78181dde6cfc146c09f55e29e
commit-date: 2023-03-02
host: x86_64-unknown-linux-gnu
release: 1.69.0-nightly
LLVM version: 15.0.7
```

**Comment by dtolnay** [comment]
I suggested a possible generalization of this in https://github.com/rust-lang/rust-clippy/issues/10463.

**Comment by blyxyas** [comment]
Should this issue be closed, as #10463 has been closed?

**Comment by dtolnay** [comment]
:+1:
